module/VuFind/src/VuFind/Autocomplete/Solr.php

```diff
@@ -171,21 +178,51 @@ protected function initSearchObject()
     /**
      * Process the user query to make it suitable for a Solr query.
      *
-     * @param string $query Incoming user query
+     * @param string $query  Incoming user query
+     * @param ?array $extras Array of extra parameter
      *
-     * @return string       Processed query
+     * @return string        Processed query
      */
-    protected function mungeQuery($query)
+    protected function mungeQuery(string $query, array $extras = null): string
     {
         // Modify the query so it makes a nice, truncated autocomplete query:
         $forbidden = [':', '(', ')', '*', '+', '"', "'"];
         $query = str_replace($forbidden, ' ', $query);
-        if (!str_ends_with($query, ' ')) {
+        if (
+            !str_ends_with($query, ' ') && (!isset($extras[self::NO_WILDCARD]) || $extras[self::NO_WILDCARD] !== true)
```

Comment:
This might be more readable (or at least more concise) as:

```suggestion
            !str_ends_with($query, ' ') && !($extras[self::NO_WILDCARD] ?? false))
```